### PR TITLE
Index State Management support

### PR DESCRIPTION
**Pull request by florianvazelle**
### Description
Added support for the ISM API.

### Issues Resolved
The issue of managing index states from the python client.  
This issue is the subject of a feature request at https://github.com/opensearch-project/opensearch-py/issues/284.

### CODE REVIEW

**Comment by florianvazelle** [comment]
Hi @saimedhi, some tests seems to start opensearch without ISM plugin. Is this normal ?

https://github.com/opensearch-project/opensearch-py/actions/runs/5333494406/jobs/9664033038

**Comment by saimedhi** [comment]
Hi @florianvazelle, I am not completely sure but I suggest to take a look at other existing plugin tests. I have gone through Alerting plugin tests. They are passing for "integration" tests and skipped for "Integration with Unreleased Opensearch" tests. So, please take a look at those tests once. Thanks :)

File: test_opensearchpy/test_server/test_plugins.py [hunk 1]
```diff
@@ -181,3 +182,107 @@ def test_run_monitor(self):
         self.assertIn("monitor_name", response)
         self.assertIn("period_start", response)
         self.assertIn("period_end", response)
+
+
+class TestIndexManagementPlugin(OpenSearchTestCase):
```
Comment (dblock) [inline test_opensearchpy/test_server/test_plugins.py:187]: We should break tests for each plugin in their own file(s), this is going to get really big.

**Review by dblock — CHANGES_REQUESTED** [review]
Small request to move plugin tests into separate file(s). Please do for existing ones if this isn't the case.

**Comment by dblock** [comment]
Looks good, needs a rebase + user guides.

**Comment by florianvazelle** [comment]
@dblock done ! :slightly_smiling_face:

File: USER_GUIDE.md [hunk 2]
```diff
@@ -21,6 +21,10 @@
       - [**Creating a destination**](#creating-a-destination)
       - [**Getting alerts**](#getting-alerts)
       - [**Acknowledge alerts**](#acknowledge-alerts)
```
Comment (dblock) [inline USER_GUIDE.md:23]: Mind fixing these `**` in another PR? Thanks.

**Review by dblock — APPROVED** [review]

**Pull request merged**